ykrt/src/mt.rs

```diff
@@ -59,10 +60,32 @@ static SERIALISE_COMPILATION: LazyLock<bool> = LazyLock::new(|| {
     &env::var("YKD_SERIALISE_COMPILATION").unwrap_or_else(|_| "0".to_owned()) == "1"
 });
 
+#[derive(Debug, Copy, Clone)]
+pub struct SideTraceInfo {
```

Comment:
Let's always use `pub(crate)` unless we *have* to make things `pub`.

Reply:
```
warning: type `SideTraceInfo` is more private than the item `Compiler::compile`                                                                                                                                                         
  --> ykrt/src/compile/mod.rs:26:5                                                                                                                                                                                                      
   |                                                                                                                                                                                                                                    
26 | /     fn compile(                                                                                                                                                                                                                  
27 | |         &self,                                                                                                                                                                                                                   
28 | |         irtrace: MappedTrace,                                                                                                                                                                                                    
29 | |         sti: &SideTraceInfo,                                                                                                                                                                                                     
30 | |     ) -> Result<(*const c_void, Option<NamedTempFile>), Box<dyn Error>>;                                                                                                                                                         
   | |________________________________________________________________________^ method `Compiler::compile` is reachable at visibility `pub`                                                                                             
   |                                                                                                                                                                                                                                    
note: but type `SideTraceInfo` is only usable at visibility `pub(crate)`                                                                                                                                                                
  --> ykrt/src/mt.rs:64:1                                                                                                                                                                                                               
   |                                                                                                                                                                                                                                    
64 | pub(crate) struct SideTraceInfo {                                                                                                                                                                                                  
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                    
   = note: `#[warn(private_interfaces)]` on by default
```

Reply:
OK.